### PR TITLE
Make GetSpatialIdsWithinRadiusOfLine() return expected result when radius==0

### PR DESCRIPTION
**Pull request by buscandoaverroes**
address #15 

Edits:
- add `radius==0` conditional to GetSpatialIdsWithinRadiusOfLine()
- add `clearance==0` conditional to  FitClearanceAroundExtendedSpatialID()
- correct expected values for TestFitClearanceAroundExtendedSpatialID01()
  - clearance=0 should return 0
- add log files to .gitignore (stop tracking)
 
Add:
- TestGetSpatialIdsWithinRadiusOfLine(): tests GetSpatialIdsWithinRadiusOfLine() when radius==0
- TestFitClearanceAroundExtendedSpatialID05() tests FitClearanceAroundExtendedSpatialID when clearance>0 but clearance<1 

Memo: as of [58638b9](https://github.com/trajectoryjp/spatial_id_go/pull/16/commits/58638b928b671281bd4f4a261d0ff4eb6ab31a83), passes package tests

### CODE REVIEW

File: transform/voxel_around_line.go [hunk 1]
```diff
@@ -42,6 +42,10 @@ func GetSpatialIdsWithinRadiusOfLine(startPoint *object.Point, endPoint *object.
 		return nil, error
 	}
 
+	if radius == 0 {
```
Comment (HarutakaMatsumoto) [inline transform/voxel_around_line.go:45]: This does not apply to a minute radius like 0.01.
The cause is that hLayers and vLayers cannot be 0.
Please resolve this issue with removing this cause, without removing symptoms.

File: transform/voxel_around_line.go [hunk 2]
```diff
@@ -179,9 +183,14 @@ func GetSpatialIdsWithinRadiusOfLine(startPoint *object.Point, endPoint *object.
...
 func FitClearanceAroundExtendedSpatialID(spatialID string, clearance float64) (horizontalLayer int64, verticalLayer int64, error error) {
 
-	// validate clearance
+	// validate clearance: There are two special cases:
+	// 1. The clearance must be non-negative (clearance must be >= 0)
+	// 2. If the clearance is exactly 0, return 0 for both the horizontalLayer and verticalLayer. This is because
+	// if clearance is 0, the only Spatial IDs that are on the route line should be used -- no surrounding Spatial IDs
 	if clearance < 0 {
 		return 0, 0, fmt.Errorf("\ninvalid clearance value. Clearance must be >= 0")
+	} else if clearance == 0 {
```
Comment (HarutakaMatsumoto) [inline transform/voxel_around_line.go:192]: This does not apply to a minute radius like 0.01.
The cause is that hLayer and vLayer cannot be 0.
Please resolve this issue with removing this cause, without removing symptoms.

File: transform/voxel_around_line_test.go [hunk 3]
```diff
@@ -13,6 +13,44 @@ import (
 	"github.com/trajectoryjp/spatial_id_go/v2/shape"
 )
 
+func TestGetSpatialIdsWithinRadiusOfLine(t *testing.T) {
```
Comment (HarutakaMatsumoto) [inline transform/voxel_around_line_test.go:16]: This test will pass without merging this pull request.
Modify this test and prove that the issue is resolved.

File: transform/voxel_around_line_test.go [hunk 4]
```diff
@@ -161,6 +199,30 @@ func TestFitClearanceAroundExtendedSpatialID04(t *testing.T) {
 
 }
 
+func TestFitClearanceAroundExtendedSpatialID05(t *testing.T) {
```
Comment (HarutakaMatsumoto) [inline transform/voxel_around_line_test.go:202]: This test will pass without merging this pull request.
Modify this test and prove that the issue is resolved.

**Review by HarutakaMatsumoto — CHANGES_REQUESTED** [review]
I reviewed this.
Check it.

File: transform/voxel_around_line_test.go (hunk 3 above)
Comment (buscandoaverroes) [reply transform/voxel_around_line_test.go:16]: when GetSpatialIdsWithinRadiusOfLine(..., skipsMeasurement=true) the test fails at commit 46c3a4103867fbf9099f42f2d70d18f401b40126

File: transform/voxel_around_line.go (hunk 2 above)
Comment (buscandoaverroes) [reply transform/voxel_around_line.go:192]: addressed in 4e63fe7b827eac8b75776a6e04041f3e4a830e77 .
hLayer and vLayer can now be returned as 0

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 **/.DS_Store
 .vscode/**/*
+**/log/**/*
```
[the remark this answers is not in the document]
Comment (buscandoaverroes) [reply .gitignore]: reverted in https://github.com/trajectoryjp/spatial_id_go/pull/16/commits/3641beded29b5a534d7eba5494497b455df4d2ce

File: transform/voxel_around_line_test.go (hunk 4 above)
Comment (buscandoaverroes) [reply transform/voxel_around_line_test.go:239]: TestFitClearanceAroundExtendedSpatialID01 should fail before pull request https://github.com/trajectoryjp/spatial_id_go/blob/58638b928b671281bd4f4a261d0ff4eb6ab31a83/transform/voxel_around_line_test.go#L106

**Comment by buscandoaverroes** [comment]
as of [2d0b289](https://github.com/trajectoryjp/spatial_id_go/pull/16/commits/2d0b289ae5e64828297444e13d36d7034fb9056b)
- changes to .gitignore reverted
- FitClearanceAroundExtendedSpatialID() can now return `0, 0, nil`
  - GetSpatialIdsWithinRadiusOfLine() has been edited for `0, 0, nil` possibility

File: operated/shifting_spatial_id_test.go [hunk 6]
```diff
@@ -299,6 +299,63 @@ func TestGetNspatialIdsAroundVoxcels02(t *testing.T) {
 
 }
 
+func TestGetNspatialIdsAroundVoxcels03(t *testing.T) {
```
Comment (HarutakaMatsumoto) [inline operated/shifting_spatial_id_test.go:302]: Explain this test like the others.

File: operated/shifting_spatial_id_test.go [hunk 7]
```diff
@@ -299,6 +299,63 @@ func TestGetNspatialIdsAroundVoxcels02(t *testing.T) {
...
+	}
+	if !reflect.DeepEqual(map1, map2) {
+		// 戻り値の空間IDが期待値と異なる場合Errorをログに出力
+		t.Errorf("空間ID - 期待値:%v, \n取得値: %v", map1, map2)
+	}
+	t.Log("テスト終了")
+
+}
+
+func TestGetNspatialIdsAroundVoxcels04(t *testing.T) {
```
Comment (HarutakaMatsumoto) [inline operated/shifting_spatial_id_test.go:330]: Explain this test like the others.

File: transform/voxel_around_line_test.go [hunk 8]
```diff
@@ -13,6 +13,81 @@ import (
 	"github.com/trajectoryjp/spatial_id_go/v3/shape"
 )
 
+func TestGetExtendedSpatialIdsWithinRadiusOfLine01(t *testing.T) {
```
Comment (HarutakaMatsumoto) [inline transform/voxel_around_line_test.go:16]: Explain this test like the others.

File: transform/voxel_around_line_test.go (hunk 4 above)
Comment (HarutakaMatsumoto) [reply transform/voxel_around_line_test.go:239]: @buscandoaverroes 
I checked it out.
Thank you!

File: .gitignore [hunk 9]
```diff
@@ -1,2 +1,2 @@
 **/.DS_Store
-.vscode/**/*
+.vscode/**/*
```
Comment (HarutakaMatsumoto) [inline .gitignore:2]: Append a new line after this line.

**Review by HarutakaMatsumoto — APPROVED** [review]

File: transform/voxel_around_line_test.go (hunk 3 above)
Comment (HarutakaMatsumoto) [reply transform/voxel_around_line_test.go]: @buscandoaverroes
I checked it out.
Thank you!

**Comment by HarutakaMatsumoto** [comment]
I checked it out and merged it.
Thank you!

**Pull request merged**